app/Services/ScrapeService.php

```diff
@@ -0,0 +1,227 @@
+        $amenitiesFile = fopen($directoryName . $amenitiesFilename, 'a');
+
+        foreach ($building->amenities as $amenity) {
+            $record = $building->name . ';' . $amenity;
+            fwrite($amenitiesFile, $record . "\n");
+        }
+        fclose($amenitiesFile);
+    }
+
+    private function selectCountry(string $country): void {
+        $crawler = $this->client->waitFor('[name="query"]');
+        $crawler->filter('[name="query"]')->sendKeys($country);
+        sleep(1);
+    }
+    private function fetchBuildingName(WebDriverElement&Crawler $element): string
+    {
+        return str_replace(';', ',', $element->filter('[data-testid="item-name"]')->text());
+    }
+
+    private function fetchBuildingType(WebDriverElement&Crawler $element): string
+    {
+        return str_replace(';', '', $element->filter('[data-testid="accommodation-type"]')->text());
+    }
+
+    private function fetchBuildingCity(WebDriverElement&Crawler $element): string
+    {
+        return str_replace(';', ',', $element->filter('[data-testid="distance-label-section"]')->text());
+    }
+
+    private function fetchBuildingImages(WebDriverElement&Crawler $element): array
+    {
+        $images = [];
+        $photoNum = count($element->filter("[data-testid=\"grid-gallery\"]")->children());
+        $i = 1;
+
+        while ($i <= min($photoNum, config('scraper.crawler.number_of_photos'))) {
+            $img = $element->filter("[data-testid=\"grid-image\"]:nth-of-type($i) img")->attr('src');
+
+            if (str_contains($img, ';')) {
+                continue;
+            }
+            $images[] =  $img;
+            $i++;
+        }
+
+        return $images;
+    }
+
+    private function fetchBuildingBody(WebDriverElement&Crawler $element): string
+    {
+        $descriptionElement = $element->filter('[data-testid="accommodation-description"]');
+        return $descriptionElement->count() > 0 ? str_replace(';', ',', $descriptionElement->text()) : '';
+    }
+
+    private function fetchBuildingAddress(WebDriverElement&Crawler $element): string
+    {
+        $postalCode = $element->filter('[itemprop="postalCode"]')->text();
+        $streetAdress = $element->filter('[itemprop="streetAddress"]')->text();
+        return str_replace(";", ',', $streetAdress . $postalCode);
+    }
+
+    private function fetchBuildingAmenities(WebDriverElement&Crawler $element): array
+    {
+        $amenities = [];
+
+        try {
+            $element->filter('details ul')
+                ->each(function (Crawler $amenitiesList) use (&$amenities) {
+                    $amenitiesList->filter('li')
+                        ->each(function (Crawler $amenity) use (&$amenities) {
+                            $amenities[] = str_replace(';', ',', $amenity->text());
+                        });
+                });
+        } catch (Exception) {
+            echo 'error in amenities';
+        }
+        return $amenities;
```

Comment:
I just left a comment on this part because there is still an unresolved thread on this in the previous code review.